lib/AST/ASTPrinter.cpp

```diff
@@ -5050,6 +5098,13 @@ void PrintAST::visitErasureExpr(ErasureExpr *expr) {
 }
 
 void PrintAST::visitKeyPathExpr(KeyPathExpr *expr) {
+  auto components = expr->getComponents();
```

Comment:
You will need to print the root type (`getRootType()`) before the components, e.g `\<RootType>.<component1>.<component2>`

Reply:
Not as important, but it would also be nice if this could handle printing key paths before the components are resolved. In that case, the components will be empty, and you'll want to print the `getParsedRoot()` and `getParsedPath()` instead.

Reply:
You mean to say that if `components` is `empty`, just after the `isObjC()`, we need to print `ParsedRoot` and `ParsedPath`?

Reply:
Yes